fluent/exceptions.py

```diff
@@ -0,0 +1,15 @@
+from __future__ import absolute_import, unicode_literals
+
+
+class FluentFormatError(ValueError):
+    def __eq__(self, other):
+        return ((other.__class__ == self.__class__) and
+                other.args == self.args)
+
+
+class FluentReferenceError(FluentFormatError):
+    pass
+
+
+class FluentCyclicReferenceError(FluentFormatError):
```

Comment:
Is there a value in specializing the errors? I would think that generic `ParserError` and `ResolverError` would serve us quite far, and in your case I see you always using the string value of the error anyway for cyclic reference.
Not strongly opinionated, happy to keep it if you think it fits python better.

Reply:
My instinct is to leave these as specialized. I've never had a problem from overly specialized exceptions, but seen plenty of problems from overly broad - for anyone who is interested in certain types of exceptions, they end up having to look at the message itself, which is fragile, instead of the exception type. Plenty of good Python libraries define a plethora of exception types - e.g [SqlAlchemy exceptions](https://docs.sqlalchemy.org/en/latest/core/exceptions.html)